From 2918f0daaef7f8628bc045a011e9400345f08024 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Mon, 13 Dec 2021 07:34:19 +0200
Subject: [PATCH 20/20] s32gen1: sram: Report the size of the cleared SRAM

This corrects the status message printed after the SRAM
initialization. It prints the size of the cleared SRAM
region instead of the operation status.

Issue: ALB-8271
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/cpu/armv8/s32/initsram.c | 14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)

diff --git a/arch/arm/cpu/armv8/s32/initsram.c b/arch/arm/cpu/armv8/s32/initsram.c
index 2a70e00e14..e5fbdd9dc7 100644
--- a/arch/arm/cpu/armv8/s32/initsram.c
+++ b/arch/arm/cpu/armv8/s32/initsram.c
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier:     GPL-2.0+
 /*
- * Copyright 2018,2020 NXP
+ * Copyright 2018,2020-2021 NXP
  */
 
 #include <asm/arch-s32/soc.h>
@@ -13,7 +13,7 @@ static int do_init_sram(cmd_tbl_t *cmdtp, int flag, int argc,
 			char * const argv[])
 {
 	unsigned long addr;
-	unsigned size, max_size, ret_size;
+	unsigned int size, max_size, ret;
 	char *ep;
 
 	if (argc < 3)
@@ -52,22 +52,18 @@ static int do_init_sram(cmd_tbl_t *cmdtp, int flag, int argc,
 	}
 
 	invalidate_dcache_range(addr, addr + size);
-	ret_size = sram_clr(addr, size);
-
-	if (!ret_size) {
+	ret = sram_clr(addr, size);
+	if (!ret) {
 		printf("Init SRAM failed\n");
 		return CMD_RET_FAILURE;
 	}
 
-	printf("Init SRAM region at address 0x%08lX, size 0x%X bytes ...\n",
-	       addr, ret_size);
-
 	return CMD_RET_SUCCESS;
 }
 
 U_BOOT_CMD(
 		initsram,	3,	1,	do_init_sram,
-		"DMA init SRAM from address",
+		"Initialize SRAM from address",
 		"startAddress[hex] size[hex]"
 	  );
 
-- 
2.17.1

